sips/sip-015/sip-015-network-upgrade.md

```diff
@@ -0,0 +1,1595 @@
+  re-lock for some subsequent number of reward cycles while still
+  being locked.
+    - This would also be true for `pox-2.delegate-stack-extend`
+- Calls to PoX 2 which would attempt to create state for a cycle
+  _before_ `(N+1)` will fail
+- Calls to the original PoX contract which would attempt to create
+  state for a cycle `>= N+1` will be made to fail and any state after
+  `N+1` is ignored.  This requires interposing on contract-calls
+  during period 2 and checking the reward cycles arguments: the
+  relevant functions are `stack-stx`, `delegate-stack-stx`,
+  `stack-aggregation-commit`, and `reject-pox`.
+
+### New method: `stack-extend`
+
+This method allows direct stackers to re-lock their funds for up to an additional
+12 cycles *before* their current lock has expired. The target use case for
+this is to allow users to repeatedly stack without a cooldown phase.
+
+This method checks that the stacker is still allowed to stack, that
+the stacker has not delegated to an operator, and that their funds are
+currently stacked. The caller may supply a new reward address for the
+extension.
+
+The special case handler for the PoX contract in the Clarity VM will
+check this method's return value and set the stacker's STX account to
+"auto-unlock" at the end of the last extended-to reward cycle.
+
+### New method: `delegate-stack-extend`
+
+This method allows operators to re-lock the funds of one of their
+delegation participants for additional cycles *before* that participants
+lock has expired. The target use case for this is to allow delegation
+operators to repeatedly stack on behalf of their users without cooldowns.
+
+This method checks that the delegator is still authorized on behalf of
+the given stacker (and remains authorized until the end of the
+locked-for period) and that the funds are currently stacked.
+
+Note that, just as with the existing `delegate-stack-stx` function,
+this method locks *but does not commit* the user's STX. The delegation
+operator must invoke `stack-aggregation-commit` to set a reward address
+for the locked funds.
+
+The special case handler for the PoX contract in the Clarity VM will
+check this method's return value and set the stacker's STX account to
+"auto-unlock" at the end of the last extended-to reward cycle.
+
+### New method: `stack-increase`
+
+This method allows direct stackers to lock additional funds for
+the remaining cycles on their current lock.
+
+This method checks that the stacker is still allowed to stack, that
+the stacker has not delegated to an operator, that their funds are
+currently stacked, and that they have enough unlocked funds to cover
+the increase. The caller may *not* supply a new reward address for the
+increase.
+
+The special case handler for the PoX contract in the Clarity VM will
+check this method's return value and set the locked amount in the
+stacker's STX account to correspond to the increased amount.
+
+### New method: `delegate-stack-increase`
```

Comment:
it should be `delegate-stack-stx-increase` to distinguish from a new `delegate-stx-increase' function.